emulator.go

```diff
@@ -247,6 +246,10 @@ func (s *Server) CreateTask(ctx context.Context, in *tasks.CreateTaskRequest) (*
 		return nil, status.Errorf(codes.FailedPrecondition, "The queue no longer exists, though a queue with this name existed recently.")
 	}
 
+	if (in.Task.Name != "") && !isValidTaskName(in.Task.Name) {
```

Comment:
Why the need for the additional `in.Task.Name != ""`?

Reply:
It can be empty in which case the emulator/GTasks allocates a random id. That's probably the most common case, as the docs advise that specifying your own name increases latency significantly.

I could maybe rename the function `isValidNameForNewTask` or something, and move the empty check into the func (with explanatory comment) to simplify that `if` if you want?

Reply:
No that's fine, I read the `and` as an `or` for some reason.